indexing-service/src/main/java/org/apache/druid/indexing/overlord/setup/FillCapacityWithCategorySpecWorkerSelectStrategy.java

```diff
@@ -32,13 +33,17 @@
 public class FillCapacityWithCategorySpecWorkerSelectStrategy implements WorkerSelectStrategy
 {
   private final WorkerCategorySpec workerCategorySpec;
+  private final TaskLimits taskLimits;
+
 
   @JsonCreator
   public FillCapacityWithCategorySpecWorkerSelectStrategy(
-      @JsonProperty("workerCategorySpec") WorkerCategorySpec workerCategorySpec
+      @JsonProperty("workerCategorySpec") WorkerCategorySpec workerCategorySpec,
+      @JsonProperty("taskLimits") @Nullable TaskLimits taskLimits
   )
   {
     this.workerCategorySpec = workerCategorySpec;
+    this.taskLimits = Configs.valueOrDefault(taskLimits, new TaskLimits());
```

Comment:
Rather than using `new TaskLimits()`, I would advise keeping a `TaskLimits.EMPTY` instance that can be used everywhere.